hygieia-jenkins-plugin-alt/src/main/hygieia/HygieiaJenkinsPluginAlt.groovy

```diff
@@ -0,0 +1,170 @@
+    }
+
+    def payloadData = parseJson(response.content)
+    
+	if (payloadData.changeSets.items.author.fullName[0]!=null) {
+        def developer = payloadData.changeSets.items.author.fullName[0]
+        scmCommitDeveloper = developer[0]
+    }
+    
+    if (payloadData.changeSets.items.msg[0]!=null) {
+        def changedDescription = payloadData.changeSets.items.msg[0]
+        scmFileChangedDescription = changedDescription[0]
+    }
+	
+    if (payloadData.changeSets.items.affectedPaths[0]!=null) {
+        def fileChanged = payloadData.changeSets.items.affectedPaths.size()
+        noOfFilesChanged = fileChanged
+    }
+}
+
+
+def parseJson(text) {
+    return new JsonSlurperClassic().parseText(text)
+}
+
+// Function to post deployment metrics to Hygieia
+def postMetrics(authtoken, hygieiaApiUrl, number, buildUrl, jobName, buildStatus, startTime, jobUrl, instanceUrl, endTime, duration, url, branch, type, scmRevisionNumber, scmCommitTimestamp, artifactName, canonicalName, artifactGroup, artifactVersion, artifactModule, artifactExtension, buildId, timestamp, appName, envName, deployStatus) {
+	getScmMetrics(buildUrl, authtoken)
+	postHygieiaBuildMetrics(hygieiaApiUrl, number, buildUrl, jobName, buildStatus, startTime, jobUrl, instanceUrl, endTime, duration, url, branch, type, scmRevisionNumber, scmCommitTimestamp)
+	postHygieiaArtifactMetrics(hygieiaApiUrl, artifactName, canonicalName, artifactGroup, artifactVersion, artifactModule, artifactExtension, buildId, timestamp, scmRevisionNumber, scmCommitTimestamp, jobName, url, branch, buildUrl, jobUrl, number, instanceUrl)
+	postHygieiaDeployMetrics(hygieiaApiUrl, number, jobUrl, appName, envName, artifactName, artifactVersion, jobName, instanceUrl, deployStatus, startTime, endTime, duration)
+}
+
+
+def postHygieiaBuildMetrics(hygieiaApiUrl, number, buildUrl, jobName, buildStatus, startTime, jobUrl, instanceUrl, endTime, duration, giturl, branch, type, scmRevisionNumber, scmCommitTimestamp) {
+	def customHeaders = [[name: 'Content-Type', value: 'application/json']]
+	hygieiaApiUrl = hygieiaApiUrl + "build"
+	def contents = """
+	{  
+	   "number":"$number",
+	   "buildUrl":"$buildUrl",
+	   "jobName":"$jobName",
+	   "buildStatus":"$buildStatus",
+	   "startTime": "$startTime",
+	   "jobUrl":"$jobUrl",
+	   "instanceUrl":"$instanceUrl",
+	   "niceName":"$jobName",
+	   "endTime":"$endTime",
+	   "duration":"$duration",
+	   "codeRepos":[  
+		  {  
+			 "url":"$giturl",
+			 "branch":"$branch",
+			 "type":"$type"
+		  }
+	   ],
+	   "sourceChangeSet":[  
+		  {  
+			 "scmRevisionNumber":"$scmRevisionNumber",
+			 "scmCommitLog":"$scmChangedDescription",
+			 "scmAuthor":"$scmCommitDeveloper",
+			 "scmCommitTimestamp":"$scmCommitTimestamp",
+			 "numberOfChanges":"$noOfFilesChanged"
+		  }
+	   ]
+	}"""
+    def response = httpRequest httpMode: 'POST', requestBody: contents, customHeaders: customHeaders, url: hygieiaApiUrl
```

Comment:
Should the hygieiaApiUrl be defined in the Jenkins global properties or at the pipeline step? Also do you need supporting apis for the uri /deploy and /artifact for this client to work?

Reply:
Build, artifact and deploy api's are available within Hygieia API. HygieiaApiUrl should point to your hygieia api, url will be like - https://hygieiaapihost:port/api/build, https://hygieiaapihost:port/api/artifact and https://hygieiaapihost:port/api/deploy. No additional supporting api's are need. I recommend using pipelineascode - Jenkinsfile and declare the hygieiaApiUrl within that file.

Reply:
ok, however the version of apis invoked seemed to be little dated. However lets leave the discretion to users to choose the endpoints. Can you update the documentation for these as well?